central/scannerdefinitions/handler/handler.go

```diff
@@ -381,31 +395,43 @@ func (h *httpHandler) openMostRecentDefinitions(ctx context.Context, uuid string
 	return
 }
 
-func (h *httpHandler) openMostRecentMappings(fileName string) (file *vulDefFile, err error) {
+func (h *httpHandler) openMostRecentFile(updaterKey string, fileName string) (file *vulDefFile, err error) {
 	// TODO(ROX-20520): enable fetching offline file
-	log.Info("Getting v4 repository mapping data updater")
-	u := h.getUpdater(mappingUpdaterKey)
-
+	log.Debugf("Getting v4 data for updater key: %s", updaterKey)
+	u := h.getUpdater(updaterKey)
 	var onlineFile *vulDefFile
-	onlineZipFile, onlineTime, err := h.startUpdaterAndOpenFile(u)
-	if err != nil || onlineZipFile == nil {
-		return
+	// Ensure the updater is running.
+	u.Start()
+	openedFile, onlineTime, err := u.file.Open()
+	if err != nil {
+		return nil, err
 	}
-	log.Infof("Compressed repository mapping file is available: %s", onlineZipFile.Name())
+	if openedFile == nil {
+		return nil, errors.New("No valid file can be opened")
+	}
+	log.Debugf("Compressed data file is available: %s", openedFile.Name())
 	toClose := func(f *vulDefFile) {
 		if file != f && f != nil {
 			utils.IgnoreError(f.Close)
 		}
 	}
-	targetFile, err := openFromArchive(onlineZipFile.Name(), fileName)
-	if err != nil {
-		return nil, err
-	}
-	if targetFile != nil {
-		onlineFile = &vulDefFile{File: targetFile, modTime: onlineTime}
+	switch updaterKey {
+	case cvssUpdaterKey:
+		onlineFile = &vulDefFile{File: openedFile, modTime: onlineTime}
+	case mappingUpdaterKey:
+		targetFile, err := openFromArchive(openedFile.Name(), fileName)
+		if err != nil {
+			return nil, err
+		}
+		if targetFile != nil {
+			onlineFile = &vulDefFile{File: targetFile, modTime: onlineTime}
+		} else {
+			return nil, fmt.Errorf("cannot find associated mapping file: %s", fileName)
+		}
+	default:
+		return nil, fmt.Errorf("fail to get updater and file: %s", updaterKey)
```

Comment:
```suggestion
		return nil, fmt.Errorf("internal error: updater key: %s", updaterKey)
```

(I think this is an internal error because the updater was already retrieved from the updater mapping.)